reuse Dimension::TYPE_ consts instead of introducing new one for metric semantic types

diosmosis opened the pull request:
### Description:

I noticed the values for the SEMANTIC_TYPE_... consts I added in #20415 already exist as `Dimension::TYPE_` so rather than add more to the API it seems a better idea to reuse these values. They exist for the same purpose, just for the dimension column in reports, so I think it's the better approach.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

diosmosis committed:
reuse Dimension::TYPE_ consts instead of introducing new one for metric semantic types

diosmosis committed:
Merge branch '4.x-dev' into reuse-dimension-type-values

diosmosis committed:
update submodule

diosmosis committed:
updating expected test files

sgiehl commented on core/Columns/Dimension.php:
```diff
@@ -49,6 +49,7 @@ abstract class Dimension
     const TYPE_TIMESTAMP = 'timestamp';
     const TYPE_BOOL = 'bool';
     const TYPE_PERCENT = 'percent';
+    const TYPE_UNKNOWN = 'unknown';
```
I don't think we should add a dimension type `unknown`. A dimension that is defined within Matomo can't logically have that type.

diosmosis replied:
It just means unformatted text. I guess it could also be replaced w/ just `text`.

sgiehl replied:
in terms of dimensions I think text makes more sense.

sgiehl commented on plugins/CustomDimensions/tests/System/expected/test___API.getReportMetadata_day.xml:
```diff
@@ -29,7 +29,7 @@
 			<nb_visits>number</nb_visits>
 			<nb_actions>number</nb_actions>
-			<avg_time_on_site>duration</avg_time_on_site>
+			<avg_time_on_site>duration_s</avg_time_on_site>
```
the change in this type name is ok for Looker Studio? Same for money <-> currency.

diosmosis replied:
The name values are irrelevant, they can easily be converted to what is expected within Looker Studio.

sgiehl reviewed (changes requested):
I'm not fully sure if we should mix up using dimension types and semantic types. If I understood that correctly those semantic type were mainly added as additional metadata for looker studio.
I'm not very familiar with that, so I'm unable to say if Looker Studio would be able to work with all dimension types we currently have defined. If it only supports a fixed set of types I think it might be better to define them separately. So we don't break anything when adding/changine new dimension types.

diosmosis commented:
> I'm not fully sure if we should mix up using dimension types and semantic types. If I understood that correctly those semantic type were mainly added as additional metadata for looker studio.

Yes, they were added for additional metric metadata, saying how metric values should be formatted and what they actually contain. Which is exactly what the Dimension::TYPE_* values do as well. (Otherwise I can't see why there would be a `percent` value.)

sgiehl commented:
If those dimension types can be easily used for that purpose, we should for sure use them. Was just unsure because of the use in Looker Studio. But If that isn't a problem, then it's fine to change that.

diosmosis committed:
remove Dimension::TYPE_UNKNOWN and use "unspecified" in report metadata output if an empty value is provided + support metric semantic type in ArchivedMetric/ComputedMetric for CustomReports

diosmosis committed:
update expected screenshots

diosmosis committed:
fix integration test

diosmosis commented:
@sgiehl changed the PR to use `null` if a metric type is unspecified, and changed the `unknown` text to `unspecified` (only for metric metadata, not for dimension metadata). Also added semantic type support to ArchivedMetric/ComputedMetric which were already using the `Dimension::TYPE_` values. Let me know your thoughts.

diosmosis committed:
remove unused imports

sgiehl committed:
submodule updates
